File: examples/python/01_simple_completion.py
"""
Simple Completion Example

Demonstrates the basic usage of ModelSuite for making completion requests.

Requirements:
- Set ANTHROPIC_API_KEY environment variable (or another provider's key)

Run:
    python 01_simple_completion.py
"""

from modelsuite import ModelSuiteClient, Message, CompletionRequest


def main():
    # Create client from environment variables
    # Automatically detects configured providers
    client = ModelSuiteClient.from_env()

    # List available providers
    providers = client.providers()
    print(f"Available providers: {providers}")

    # Create a simple completion request
    # Use "provider/model" format for explicit provider routing
    request = CompletionRequest(
        model="anthropic/claude-sonnet-4-20250514",
        messages=[
            Message.user("What is the capital of France? Reply in one word.")
        ],
        max_tokens=100,
    )

    # Make the request
    print("\nSending request...")
    response = client.complete(request)

    # Print the response
    print(f"\nResponse: {response.text_content()}")
    print(f"Model: {response.model}")
    print(f"Stop reason: {response.stop_reason}")

    # Print token usage
    if response.usage:
        print(f"\nToken usage:")
        print(f"  Input tokens: {response.usage.input_tokens}")
        print(f"  Output tokens: {response.usage.output_tokens}")
        print(f"  Total tokens: {response.usage.total_tokens()}")


if __name__ == "__main__":
    main()

File: examples/nodejs/01-simple-completion.ts
/**
 * Simple Completion Example
 *
 * Demonstrates the basic usage of LLMKit for making completion requests.
 *
 * Requirements:
 * - Set ANTHROPIC_API_KEY environment variable (or another provider's key)
 *
 * Run:
 *   npx ts-node 01-simple-completion.ts
 */

import {
    JsLlmKitClient as LLMKitClient,
    JsMessage as Message,
    JsCompletionRequest as CompletionRequest,
} from 'modelsuite'

async function main() {
    // Create client from environment variables
    // Automatically detects configured providers
    const client = LLMKitClient.fromEnv()

    // List available providers
    const providers = client.providers()
    console.log('Available providers:', providers)

    // Create a simple completion request
    // Use "provider/model" format for explicit provider routing
    const request = CompletionRequest
        .create('anthropic/claude-sonnet-4-20250514', [
            Message.user('What is the capital of France? Reply in one word.')
        ])
        .withMaxTokens(100)

    // Make the request
    console.log('\nSending request...')
    const response = await client.complete(request)

    // Print the response
    console.log('\nResponse:', response.textContent())
    console.log('Model:', response.model)
    console.log('Stop reason:', response.stopReason)

    // Print token usage
    const usage = response.usage
    if (usage) {
        console.log('\nToken usage:')
        console.log('  Input tokens:', usage.inputTokens)
        console.log('  Output tokens:', usage.outputTokens)
        console.log('  Total tokens:', usage.totalTokens())
    }
}

main().catch(console.error)
